### PR TITLE
feature/2948/add-authorization-for-code-codeTrek-page

### PR DESCRIPTION
**Pull request by Hemrajbharti123**
Targets #2948 

### Description

So anyone with role employee, admin and super admin should be able to open CodeTrek page. Currently only admin and super admin has CodeTrek option in the menu
If someone is trying to open the page using link then you have implement Laravel policies like we have in invoice module so that if a user does not have permission CodeTrek.view

### Checklist:
- [x] I have performed a self-review of my own code.

### CODE REVIEW

**Comment by Hemrajbharti123** [comment]
If an unauthorized user try to open CodeTrek.view directly by using URL this error is shown to him.

image

File: Modules/CodeTrek/Http/Controllers/CodeTrekController.php [hunk 1]
```diff
@@ -20,6 +21,10 @@ public function __construct(CodeTrekService $service)
      */
     public function index()
     {
+        if (Gate::denies('codetrek_applicant.view')) {
+            abort(403, 'Unauthorized access');
+        }
+
```
Comment (GauravGusain98) [inline Modules/CodeTrek/Http/Controllers/CodeTrekController.php:27]: Lets use the approach that is used throughout the portal for policies. You can search for invoicePolicy file and see how it is working. Based on it you can implement policies using the Policy file that exists for codetrek

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comment

**Review by GauravGusain98 — APPROVED** [review]
LGTM

**Pull request merged**